nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -304,6 +304,124 @@ in {
           phone-numbers.
         '';
       };
+
+      objectstore = let
+        s3Arguments = {
+          bucket = mkOption {
+            type = types.str;
+            example = "nextcloud";
+            description = ''
+              The name of the S3 bucket.
+            '';
+          };
+          autocreate = mkOption {
+            type = types.bool;
+            description = ''
+              Create the objectstore if it does not exist.
+            '';
+          };
+          key = mkOption {
+            type = types.str;
+            example = "EJ39ITYZEUH5BGWDRUFY";
+            description = ''
+              The access key for the S3 bucket.
+            '';
+          };
+          secret = mkOption {
```

Comment:
Please remove this option.

Reply:
Can do, it will also simplify things a little. Shall we also remove `adminpass` and `dbpass` in favour of their `*File` alternatives too?

Reply:
Ideally yes. We should work toward that. A separate PR perhaps?